### PR TITLE
[Cortx 1.0] EOS-14306 CSM updates "pre-flight" checks

### PR DESCRIPTION
**Pull request by navalkp**
# Backend

## Problem Statement
<pre>
Story Ref (if any):https://jts.seagate.com/browse/EOS-14306
</pre>
## Unit testing on RPM done
<pre>
No
</pre>
## Problem Description
<pre>
Create REST API to check system status
</pre>
## Solution
<pre>
Created 2 REST API to check system status , current check status of consul and elasticsearch
GET system/status
and 

Status call for specific db
GET system/status/<db_name>    <db_name> consul , es
</pre>
## Unit Test Cases
<pre>
Tested with POSTMAN and UI tested pointing to my dev VM
</pre>
Signed-off-by:

### CODE REVIEW

File: csm/core/controllers/system_status.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    def __init__(self, request):
+        super(SystemStatusView, self).__init__(request)
+        self._service = self.request.app[const.SYSTEM_STATUS_SERVICE]
+
+    async def get(self):
+        """
+        Fetch  system status.
+        """
+        Log.debug("Handling system status request")
+        # action = self.request.match_info[const.PREFLIGHT_CONSUL]
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/system_status.py:55]: Remove the commented line

File: csm/core/controllers/system_status.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+class SystemStatusView(CsmView):
+    def __init__(self, request):
+        super(SystemStatusView, self).__init__(request)
+        self._service = self.request.app[const.SYSTEM_STATUS_SERVICE]
+
+    async def get(self):
+        """
+        Fetch  system status.
+        """
+        Log.debug("Handling system status request")
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/system_status.py:54]: Have some context in debug logs like db name

File: csm/core/services/system_status.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                Log.error(f"Status check failed for {each_resource} exception : {ex}")
+                resp[each_resource] = f"{ex}"
+                resp['success'] = False
+        return resp
+
+    async def _get_consul_status(self) -> str:
+        """
+        Return status of consul
+        """
+        Log.info("Get consul status")
```
Comment (pawankumarsrivastava) [inline csm/core/services/system_status.py:58]: Get -> Getting

File: csm/core/services/system_status.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        port = Conf.get(const.DATABASE_INDEX, 'databases.consul_db.config.port')
+        # Validation throws exception on failure
+        ConsulV().validate('service', [host, port])
+        return "success"
+
+    async def _get_elasticsearch_status(self) -> str:
+        """
+        Return status of elasticsearch
+        """
+        Log.info("Get elasticsearch status")
```
Comment (pawankumarsrivastava) [inline csm/core/services/system_status.py:70]: Get -> Getting

File: csm/core/services/system_status.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    async def check_status(self, check_list):
+        """
+        Validate status of resource
+        :param check_list: list of resources.
+        :return:
+        """
+
+        resp = dict()
+        resp['success'] = True
```
Comment (pawankumarsrivastava) [inline csm/core/services/system_status.py:42]: Add success to const file and use here

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

**Review by prathameshrodi — APPROVED** [review]

File: csm/core/controllers/system_status.py (hunk 1 above)
Comment (navalkp) [reply csm/core/controllers/system_status.py]: done

File: csm/core/controllers/system_status.py (hunk 2 above)
Comment (navalkp) [reply csm/core/controllers/system_status.py:54]: Added db_name and moved log

File: csm/core/services/system_status.py (hunk 3 above)
Comment (navalkp) [reply csm/core/services/system_status.py]: done

File: csm/core/services/system_status.py (hunk 5 above)
Comment (navalkp) [reply csm/core/services/system_status.py]: Added const and referred.

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**